rllib/policy/policy.py

```diff
@@ -764,14 +841,28 @@ def get_state(self) -> PolicyState:
             # The current global timestep.
             "global_timestep": self.global_timestep,
         }
+
+        # Add this Policy's spec so it can be retreived w/o access to the original
+        # code.
+        policy_spec = PolicySpec(
+            policy_class=type(self),
+            observation_space=self.observation_space,
+            action_space=self.action_space,
+            config=self.config,
+        )
+
         if self.config.get("enable_connectors", False):
+            state["policy_spec"] = policy_spec.serialize()
             # Checkpoint connectors state as well if enabled.
             connector_configs = {}
             if self.agent_connectors:
                 connector_configs["agent"] = self.agent_connectors.to_state()
             if self.action_connectors:
                 connector_configs["action"] = self.action_connectors.to_state()
             state["connector_configs"] = connector_configs
+        else:
+            state["policy_spec"] = policy_spec
```

Comment:
since we are breaking the compatibility here, I think we should simply always save serialized version of the spec now. like:
```
state["policy_spec"] = policy_spec.serialize()
if self.config.get("enable_connectors", False):
    ......
    state["connector_configs"] = connector_configs
```
it's safer to not save raw python objects.

Reply:
+1

Reply:
done